packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -217,15 +189,27 @@ if (!ExecutionEnvironment.canUseDOM) {
 
     let currentTime = now();
     // Next, as long as we have idle time, try calling more callbacks.
-    while (frameDeadline - currentTime > 0 && pendingCallbacks.length > 0) {
-      const latestCallbackConfig = pendingCallbacks.shift();
+    while (
+      frameDeadline - currentTime > 0 &&
+      headOfPendingCallbacksLinkedList !== null
+    ) {
+      const latestCallbackConfig = headOfPendingCallbacksLinkedList;
+      // move head of list to next callback
+      headOfPendingCallbacksLinkedList =
+        latestCallbackConfig.nextCallbackConfig;
+      if (headOfPendingCallbacksLinkedList) {
+        headOfPendingCallbacksLinkedList.previousCallbackConfig = null;
+      }
+      if (tailOfPendingCallbacksLinkedList === latestCallbackConfig) {
+        tailOfPendingCallbacksLinkedList = null;
```

Comment:
This the same as the `else` block of the previous `if` statement. If there's no next callback in the list, then the list must be empty now.

Reply:
That's true - if it's more clear will write it that way.